x-pack/platform/plugins/shared/alerting/server/backfill_client/backfill_client.ts

```diff
@@ -261,6 +271,29 @@ export class BackfillClient {
       }
     });
 
+    try {
+      for (const backfill of backfillsToSchedule) {
+        await updateGaps({
+          backfillSchedule: backfill.schedule,
+          ruleId: backfill.rule.id,
+          start: new Date(backfill.start),
+          end: backfill?.end ? new Date(backfill.end) : new Date(),
+          eventLogger,
+          eventLogClient,
+          savedObjectsRepository: internalSavedObjectsRepository,
+          logger: this.logger,
+          backfillClient: this,
+          actionsClient,
```

Comment:
It looks like we're passing the actions client through just so we can use it inside `findOverlappingBackfills` but we're also passing the backfill client which has the actions client in its context. Can we try to remove this dependency?